lib/pure/math.nim

```diff
@@ -90,7 +90,19 @@ proc isPowerOfTwo*(x: int): bool {.noSideEffect.} =
 proc nextPowerOfTwo*(x: int): int {.noSideEffect.} =
   ## returns `x` rounded up to the nearest power of two.
   ## Zero and negative numbers get rounded up to 1.
-  result = x - 1 
+
+  # Increment x if x is the smallest integer.
+  result = x
+  when defined(cpu64):
```

Comment:
Are all these conditionals really needed? Also, wouldn't it be better to test on the type, rather than the size?